p2p-tests/lib/p2p.ts

```diff
@@ -0,0 +1,228 @@
+    done();
+  }
+}
+
+export function initPeer(namespace: string): unknown | undefined {
+  const radHome = workspacePath([namespace]);
+  fs.removeSync(radHome);
+  fs.mkdirsSync(radHome);
+
+  const result = execa.sync(path.join(BIN_PATH, "radicle-proxy-init"), [
+    namespace,
+    "--key-passphrase",
+    namespace,
+    "--rad-home",
+    workspacePath([namespace]),
+  ]);
+
+  try {
+    return JSON.parse(result.stdout);
+  } catch {
+    return undefined;
+  }
+}
+
+export function startPeer(
+  namespace: string,
+  ip: string,
+  args: string[]
+): ProxyClient.ProxyClient {
+  initPeer(namespace);
+
+  const childProcess = spawnInNamespace(
+    namespace,
+    [
+      path.join(BIN_PATH, "radicle-proxy"),
+      "--peer-listen",
+      `${ip}:8776`,
+      "--http-listen",
+      `${ip}:17246`,
+      "--skip-remote-helper-install",
+      "--unsafe-fast-keystore",
+      "--insecure-http-api",
+      "--dev-log",
+      ...args,
+    ],
+    {
+      RAD_HOME: workspacePath([namespace]),
+    }
+  );
+
+  processes.push(childProcess);
+
+  return new ProxyClient.ProxyClient(`http://${ip}:17246`);
+}
+
+export function startSeed(namespace: string, args: string[]): void {
+  const radHome = workspacePath([namespace]);
+  fs.removeSync(radHome);
+  fs.mkdirsSync(radHome);
+
+  const childProcess = spawnInNamespace(
+    namespace,
+    [path.join(BIN_PATH, "upstream-seed"), "--rad-home", radHome, ...args],
+    {}
+  );
+
+  processes.push(childProcess);
+}
+
+export function spawnInNamespace(
+  namespace: string,
+  args: string[],
+  env: NodeJS.ProcessEnv
+): execa.ExecaChildProcess {
+  const subprocess = execa(
+    "ip",
+    ["netns", "exec", `upstream-test-${namespace}`, ...args],
+    {
+      env,
+    }
+  );
+
+  const stdoutPrefix = new LinePrefix(prefix(namespace));
+  const stderrPrefix = new LinePrefix(prefix(namespace));
+
+  if (subprocess.stdout) {
+    subprocess.stdout.pipe(stdoutPrefix).pipe(process.stdout);
+  }
+  if (subprocess.stderr) {
+    subprocess.stderr.pipe(stderrPrefix).pipe(process.stderr);
+  }
+
+  return subprocess;
+}
+
+export function commit(author: string, checkoutPath: string): void {
+  execa.sync("git", ["commit", "--allow-empty", "-m", "commit-message"], {
+    cwd: checkoutPath,
+    env: {
+      GIT_AUTHOR_NAME: author,
+      GIT_AUTHOR_EMAIL: `${author}@${author}.com`,
+      GIT_COMMITTER_NAME: author,
+      GIT_COMMITTER_EMAIL: `${author}@${author}.com`,
+    },
+  });
+}
+
+export function pushRad(
+  radHome: string,
+  checkoutPath: string,
+  keyPassphrase: string
+): void {
+  execa.sync("git", ["push", "rad"], {
+    cwd: checkoutPath,
+    env: {
+      RADICLE_UNSAFE_FAST_KEYSTORE: "1",
+      RAD_HOME: radHome,
+      KEY_PASSPHRASE: keyPassphrase,
+      GIT_EXEC_PATH: BIN_PATH,
+    },
+  });
+}
+
+export async function withRetry(action: () => Promise<unknown>) {
+  return await retryOnError<unknown>(action, () => true, 1000, 10);
```

Comment:
No need to specify the type parameter again.
```suggestion
  return await retryOnError(action, () => true, 1000, 10);
```

Reply:
80a72b3ce1408064d542a49d6acb4c128e732fbd